File: src/frontend/src/model/entity/Comment.ts
import {Entity} from "./Entity";
import {User} from "./User";

export interface Comment extends Entity {
    id: string;
    commentedBy: User;
    content: string;
    createdAt: string;
    commentedPostId: string;
}

export interface CreateCommentRequest {
    commentedById: string;
    commentedPostId: string;
    content: string;
}

export interface UpdateCommentRequest {
    content: string;
}

File: src/main/java/pl/marcinsalamandra/microbloggingapp/entity/Comment.java
package pl.marcinsalamandra.microbloggingapp.entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

import static javax.persistence.GenerationType.IDENTITY;

@Setter
@Getter
@NoArgsConstructor
@RequiredArgsConstructor
@Entity
@Table(name = "comments")
public class Comment {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    @Column(name = "id",
            nullable = false,
            unique = true)
    private Long id;

    @NonNull
    @ManyToOne(optional = false)
    @JoinColumn(name = "commented_by_id",
            nullable = false,
            referencedColumnName = "id")
    private AppUser commentedBy;

    @NonNull
    @ManyToOne(optional = false)
    @JoinColumn(name = "commented_post_id",
            nullable = false,
            referencedColumnName = "id")
    private Post commentedPost;

    @NonNull
    @Column(name = "content",
            nullable = false,
            length = 400)
    private String content;

    @NonNull
    @Column(name = "created_at",
            nullable = false,
            updatable = false)
    private LocalDateTime createdAt;
}
